docs/adr/0006-json-schemas-for-messaging.md

```diff
@@ -0,0 +1,92 @@
+- [Semantic Versioning][schemaver] to effectively communicate breaking vs. non-breaking message format changes
+
+## Considered Options
+
+1. Keep existing approach
+2. Use protobuf to format/validate/document message formats
+3. Use JSON-Schemas with SemVer for doc/validation of existing JSON message formats
+
+## Decision Outcome
+
+PLACEHOLDER
+
+### [option 1] Keep existing approach
+
+Currently our documentation drifts if a message format is tweaked and the docs are not updated. Ideally
+our code review process would catch this but sometimes code subleties are not caught that alter the
+message format. Validation changes can also fail to appear in the documentation, even though what is
+considered a valid message key/value should be documented.
+
+We have to occasionally duplicate `joi` schemas between repos, which should likely be moved to `fxa-shared`
+to benefit re-use, but this does not help non-Node projects consume our messages.
+
+- Pros
+  - No additional work is needed that we don't already do.
+- Cons
+  - Our documentation occasionally drifts.
+  - It's additional work to maintain accurate `joi` schemas spread throughout the code-base _or_
+  - It's additional work to consolidate our `joi` schemas in `fxa-shared`.
+
+### [option 2] Use protobuf
+
+Using protobuf would allow us to validate and document valid message formats that would also consume
+less space and be more performant to serialize/deserialize. Validation is built-in as invalid messages
+will not deserialize, and documenation can be generated based on the protobuf spec for a message.
+
+Protobuf tooling is widely available for most langauges.
+
+- Pros
+  - High throughput messaging could benefit from higher efficiency of serializing/deserializing of messages
+  - Documentation stays in sync with message spec
+  - Validation is built-in to the message serialization/deserialization process
+- Cons
+  - Messages are in binary and not easily introspectable
+  - Existing message consumers will need to be updated to handle protobuf
+  - Migration procedure will need to be created/implemented to shift from existin JSON format
+
+### [option 3] Use JSON-Schemas
+
+JSON-Schemas can define existing FxA JSON messages used in our queue system. There are libraries in
+most languages to consume these schemas for validation purposes, and libraries exist to generate
+documenation from a JSON-Schema.
+
+[SchemaVer] can be introduced to existing FxA JSON messages as consumers ignore unknown keys, while
+consumers using the JSON-Schema can verify they can handle new message formats or if they're incompatible.
+Some of this logic is manual, as [SchemaVer] is not an official part of the JSON-Schema specification.
+
+- Pros
+  - No changes needed for existing consumers
+  - Validation can be done based on the JSON-Schema without custom code
+  - Documentation can be generated to stay in sync with message spec
```

Comment:
Leaving out the second one, cause we're using Security Event Tokens rather than JSON for RPs..... except... the Security Event Token format is also JSON, so heh. :)